### DISCUSSION TITLE
Input validation callbacks are triggered 2 times for string, not for booleans

### DISCUSSION THREAD

**Discussion by gauravgs** [tiangolo/typer-cli#145] [Questions] [answered]
For the below code, I see the "str" validation callback runs twice, but for bool vars it runs only once. 
Additionally in the non-interactive CLI, i.e. passing all params with `--` in a single command, all callbacks are executed only once as expected.

```

from typing import Optional

import typer
from typing_extensions import Annotated



def version_callback(value: bool):
    print(f"Awesome CLI Version: 0.1.0")
    return value

def new_callback(value: str):
    print("In it")
    return value


def main(
    name: Annotated[str, typer.Option(prompt="Enter 'name'", callback=new_callback)],
    version: Annotated[
        bool, typer.Option(prompt="Enter 'version'", callback=version_callback)
    ],
):
    print(f"Hello {name}")
    print(version)


if __name__ == "__main__":
    typer.run(main)

```

Can someone help with this?

To reproduce this issue, run this code once as `python main.py` (you will see `new_callback` called twice) and also run it as `python main.py --name hello --version` (no extra runs)

**Comment by tiangolo** [comment, marked as answer]
Hello! Typer-CLI is now part of Typer, since version 0.12.1, so, now you should be able to install `pip install typer` and you would get the `typer` command. :rocket: 

For your use case, good question! I think I hadn't seen a callback and a prompt used together, I think it might be because of that. And it might be on the Click side. You can check that by creating an equivalent Click app, without Typer, and confirming if it behaves the same, in that case, this would be a question for Click. If not, then it is related to Typer.

If you still have problems after that, please create a new question in the Typer repo: https://github.com/tiangolo/typer/discussions/new?category=questions
